### PR TITLE
JAMES-3815 Handle possible null values in imapUidTable and messageIdTable

### PR DESCRIPTION
**Pull request by quantranhong1999**
This NPE likely comes from badly applied [Rework message denormalization migration](https://github.com/apache/james-project/blob/master/upgrade-instructions.md#rework-message-denormalization) which creates in `imapUidTable` and `messageIdTable` a few rows with null `internalDate`, `bodyStartOctet`, `fullContentOctets` and `headerContent`. Now the Cassandra driver 4 code change just triggers that NPE (can not convert a null to Date).

Checked: `bodyStartOctet`, `size`, `headerContent` should be already good from NPE.

### CODE REVIEW

**Comment by vttranlina** [comment]
image
Should we fix the NPE of other classes in this pr?

**Comment by quantranhong1999** [comment]
> Should we fix the NPE of other classes in this pr?

IMO it depends on which tables and do we allow to write null into that. I will check tmr.

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java [hunk 1]
```diff
@@ -538,4 +540,36 @@ private ThreadId getThreadIdFromRow(Row row, MessageId messageId) {
         }
         return ThreadId.fromBaseMessageId(CassandraMessageId.Factory.of(threadIdUUID));
     }
+
+    @VisibleForTesting
+    Mono<Void> insertNullInternalDateAndHeaderContent(CassandraMessageMetadata metadata) {
```
Comment (chibenwa) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java:545]: Imo overkill

**Review by chibenwa — COMMENTED** [review]
What if size is null? HeaderContent? bodyStartOctet ?
Finally it is not a pb of badly applied migration but a regression (you) introduced during cassandra driver 4.0 upgrade.

**Comment by Arsnael** [comment]
> Finally it is not a pb of badly applied migration but a regression (you) introduced during cassandra driver 4.0 upgrade.

You mean "we"...?

**Comment by Arsnael** [comment]
> What if size is null? HeaderContent? bodyStartOctet ? 

Answer in his first comment of the issue: 

```
Checked: bodyStartOctet, size, headerContent should be already good from NPE.
```

**Review by Arsnael — COMMENTED** [review]
Read it looks ok to me so far

**Comment by quantranhong1999** [comment]
> What if size is null? HeaderContent? bodyStartOctet ?

We already wrap `Optional.ofNullable` when reading them. 
Even `bodyStartOctet` and `size` are special cases when the driver does not return the `null` value but defaults 0 instead. May be return actual `null` here will be more exact.

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**